plugins/TimingEndpointController.cpp

```diff
@@ -51,20 +52,14 @@ TimingEndpointController::TimingEndpointController(const std::string& name)
 void
 TimingEndpointController::do_configure(const nlohmann::json& data)
 {
-  auto conf = data.get<timingendpointcontroller::ConfParams>();
```

Comment:
I really don't know what you are trying to do here! The old code was reading values from configuration and storing them in member variables. Why does the new code not do the same (or do it in `init()`)?

Reply:
I was trying to use the io manager function to do something similar as was done in listrev, apologies! I think I did implement the reading of the configs in the init method but in `TimingController.cpp` as that's the only place that had an init method which I thought all the plugins may be able to read from? should I just copy the relevant bits from `TimingController.cpp `'s init and create an init function in here?

Reply:
There is more than one way to do things but I would be inclined to change `TimingController::init()` to just save a copy of the `appfwk::ModuleConfiguration` pointer to a protected member variable and have each `do_configure` method extract the configuration it requires through that pointer.